src/tribler/gui/widgets/downloadspage.py

```diff
@@ -347,46 +346,56 @@ def on_remove_download_clicked(self, checked):
 
     def on_remove_download_dialog(self, action):
         if action != 2:
-            for selected_item in self.selected_items:
-                infohash = selected_item.download_info["infohash"]
+            for item in self.selected_items:
                 current_download = self.window().download_details_widget.current_download
-                if current_download and current_download.get(infohash) == infohash:
```

Comment:
From my POV the code ` if current_download and current_download.get(infohash)` was incorrect.